daos-stack/daos
# DAOS-8310 test: add ior performance tests

Skip-unit-tests: true
skip-test-rpms: true
Test-tag: performance,-manual ior,pr

- Add ftest/performance with performance tests
  - performance test base
  - IOR Easy DFS SX
  - IOR Easy DFS EC_16P2GX
  - IOR Easy Dfuse SX
  - IOR Easy Dfuse EC_16P2GX
  - IOR Easy DFS EC_4P2GX, stop rank during write
  - IOR Easy DFS EC_4P2GX, stop rank during read
  - IOR Easy DFS EC_16P2GX, stop rank during write
  - IOR Easy DFS EC_16P2GX, stop rank during read
  - IOR Easy HDF5 SX
  - IOR Easy MPIIO SX
  - IOR Hard DFS SX
  - IOR Hard DFS EC_16P2GX
  - IOR Hard Dfuse SX
  - IOR Hard Dfuse EC_16P2GX

- Add IorCmd.sw_deadline
- Adjust IorCmd.get_ior_metrics to accept string input

- Add oclass_utils.py

- Add DaosServerManager.get_daos_metrics
  - Calls daos_metrics for each engine.

- Add SubprocessManager.get_expected_states
  - Returns the expected states for ranks

- Add DaosServerManager.stop_random_rank
  - Stops a random running rank

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>
